src/librustdoc/html/tests.rs

```diff
@@ -0,0 +1,43 @@
+use crate::html::format::href_relative_parts;
+#[test]
+fn href_relative_parts_basic() {
+    let relative_to_fqp: Vec<String> = ["std", "vec"].iter().copied().map(String::from).collect();
+    let fqp: Vec<String> = ["std", "iter"].iter().copied().map(String::from).collect();
+    let expected = vec!["..", "iter"];
+    assert_eq!(expected, href_relative_parts(&fqp, &relative_to_fqp));
```

Comment:
nit: you use this quite a lot, maybe add a helpful function so it looks less cluttered? Something like `assert_relative_path(&["std", "vec"], &["std", "iter"], &["..", "iter"])`.